contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -0,0 +1,256 @@
+     * @param _platformAddress Address of the Curve 3pool
+     * @param _vaultAddress Address of the vault
+     * @param _rewardTokenAddress Address of CRV
+     * @param _asset Address of the supported asset
+     * @param _pToken Correspond platform token addres (i.e. 3Crv)
+     * @param _crvGaugeAddress Address of the Curve DAO gauge for this pool
+     * @param _crvMinterAddress Address of the CRV minter for rewards
+     */
+    function initialize(
+        address _platformAddress, // 3Pool address
+        address _vaultAddress,
+        address _rewardTokenAddress, // CRV
+        address _asset,
+        address _pToken,
+        address _crvGaugeAddress,
+        address _crvMinterAddress
+    ) external onlyGovernor initializer {
+        ICurvePool threePool = ICurvePool(_platformAddress);
+        for (int128 i = 0; i < 3; i++) {
+            if (threePool.coins(uint256(i)) == _asset) poolCoinIndex = i;
+        }
+        require(poolCoinIndex != -1, "Invalid 3pool asset");
+        crvGaugeAddress = _crvGaugeAddress;
+        crvMinterAddress = _crvMinterAddress;
+        InitializableAbstractStrategy._initialize(
+            _platformAddress,
+            _vaultAddress,
+            _rewardTokenAddress,
+            _asset,
+            _pToken
+        );
+    }
+
+    /**
+     * @dev Collect accumulated CRV and send to Vault.
+     */
+    function collectRewardToken() external onlyVault {
+        ICRVMinter minter = ICRVMinter(crvMinterAddress);
+        minter.mint(crvGaugeAddress);
+        IERC20 crvToken = IERC20(rewardTokenAddress);
+        uint256 balance = crvToken.balanceOf(address(this));
+        require(
+            crvToken.transfer(vaultAddress, balance),
+            "Reward token transfer failed"
+        );
+        emit RewardTokenCollected(vaultAddress, balance);
+    }
+
+    /**
+     * @dev Deposit asset into the Curve 3Pool
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     * @return amountDeposited Amount of asset that was deposited
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        onlyVault
+        returns (uint256 amountDeposited)
+    {
+        require(_amount > 0, "Must deposit something");
+        // 3Pool requires passing deposit amounts for all 3 assets, set to 0 for
+        // all
+        uint256[] memory _amounts = new uint256[](3);
+        // Set the amount on the asset we want to deposit
+        _amounts[uint256(poolCoinIndex)] = _amount;
+        // Do the deposit to 3pool
+        ICurvePool(platformAddress).add_liquidity(_amounts, uint256(0));
+        // Deposit into Gauge
+        IERC20 pToken = IERC20(assetToPToken[_asset]);
+        ICurveGauge(crvGaugeAddress).deposit(
+            pToken.balanceOf(address(this)),
+            address(this)
+        );
+        amountDeposited = _amount;
+        emit Deposit(_asset, address(platformAddress), amountDeposited);
```

Comment:
same here for deposit - the gauge will anyway emit a deposit event - so what extra benefit we'd get here

Reply:
We can collect all the events from our own contracts

Reply:
right but that's something we can build off chain and filter on our relevant addresses. The event is not free, it cost gas and with three parameters, its pricey. I defer to you on decision keep/not :+1: